mlir/test/e2e/resnet50-tosa-partition.mlir

```diff
@@ -0,0 +1,13 @@
+// RUN:   python3.7 %S/Inputs/resnet50-test.py %S/Inputs/590px-Red_Smooth_Saluki.jpg \
+// RUN: | mlir-opt --tosa-partition-pipeline --tosa-to-linalg-on-tensors --tosa-to-standard \
```

Comment:
Sanity check: is this `mlir-opt` or `miopen-opt` (that is, have we been adding the tosa partitioning stuff to our directory or to external/)?

Reply:
More significantly, per @whchung , generating test inputs at test-time is generally considered not good in LLVM - see https://github.com/ROCmSoftwarePlatform/llvm-project-mlir/blob/miopen-dialect/mlir/utils/generateInWarpTransposeTest.py and https://github.com/ROCmSoftwarePlatform/llvm-project-mlir/blob/miopen-dialect/mlir/test/Dialect/MIOpen/integration/in_warp_transpose/in_warp_transpose.mlir

Could this test be reworked to generate the MLIR offline and then to run that through the pipeline during testing? This has the advantage of letting us test the partitioning pass as well as the full end-to-end setup.